File: 206-reverse-linked-list/206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution(object):
    def reverseList(self, head):
        if not head: return None
        if not head.next: return head
        
        hold=ListNode(head.next.val, head.next.next)
        myList=ListNode(head.val)
        
        while hold!=None:
            temp=ListNode(myList.val, myList.next)
            
            myList=ListNode(hold.val, None)
            myList.next=temp
            
            hold=hold.next
        
        return myList

File: 206-reverse-linked-list/206-reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        if (head==null) return null;
        if (head.next==null) return head;
        
        ListNode hold=head.next;
        ListNode myList=new ListNode(head.val);
        
        while(hold!=null){
            ListNode temp=myList;
            
            myList=new ListNode(hold.val);
            myList.next=temp;
            
            hold=hold.next;
        }
        
        return myList;
    }
}